File: packages/python-sdk/src/treedx_sdk/adapters/observability.py
from __future__ import annotations

from typing import Any

from .common import json_request
from treedx_sdk.transport import Transport


class ObservabilityAdapter:
    def __init__(self, transport: Transport) -> None:
        self.transport = transport

    def health(self) -> Any:
        return json_request(self.transport, "GET", "/api/v1/health")

    def ready(self) -> Any:
        return json_request(self.transport, "GET", "/api/v1/ready")

    def deep_health(self) -> Any:
        return json_request(self.transport, "GET", "/api/v1/health/deep")

    def metrics(self) -> Any:
        return json_request(self.transport, "GET", "/api/v1/metrics")

File: packages/elixir-sdk/lib/treedx_sdk/adapters/observability.ex
defmodule TreeDxSdk.Observability do
  @moduledoc false
  alias TreeDxSdk.Adapters.Common

  def health(client) do
    Common.json_request(client, :get, "/api/v1/health", nil, %{})
  end

  def ready(client) do
    Common.json_request(client, :get, "/api/v1/ready", nil, %{})
  end

  def deep_health(client) do
    Common.json_request(client, :get, "/api/v1/health/deep", nil, %{})
  end

  def metrics(client) do
    Common.json_request(client, :get, "/api/v1/metrics", nil, %{})
  end
end
